### ISSUE TITLE
Feature Request / Enhancement: using lowest-price for longer than 24h period

### ISSUE THREAD

**Issue by fldutch** [ottopaulsen/node-red-contrib-power-saver#181]
If i unterstand it correctly the lowest price node searches for the given number of lowest prices in a 24h period, where i can configure the timeslots to search in in the actual 24h period?

It would be nice to extend this time to 48h - or at least to 35/36 hours.

I mainly use lowest price for charging my battery. So it would be optimal to find a given number of lowest prices within the next 35 hours, which cover the actual days power prices as well as the day ahead prices which are available around 1 p.m. here in germany. My battery is big enough to cover 2 days, so if prices today are higher then tomorrow (day ahead), it would be useful to skip charging today and charge tomorrow.

The documentation says i could only use 2 digits for the dynamic config using fromTime/toTime from 00-23. If it is also be possible to enter times including the date, i might give it a try and no enhancement would be neccessary.

Yes, i might also be able to handle this otherwise :).

**Comment by ottopaulsen** [comment]
This has been suggested before: #35 

The logic to make this work is not trivial.
For example:
You want to select the 6 cheapest hours during a 34 hour period from 14:00 today until the end of the next day. Maybe the cheapest hours are the 6 last tomorrow.
Then tomorrow at 13:15 you get prices for the next day. Your 6-hour period has not started yet, and the price is going down, so the cheapest now is the day after. Should you postpone another day? And what if there was 2 cheap hours earlier, that you have used, then what?

As you may understand, this issue creates lots of questions that must be asked, and it will require additional rules that must be defined and implemented.

So, until someone can come up with a complete suggestion with rules that are understandable and that makes sense, I just cannot implement this feature. Sorry :-)

**Comment by fldutch** [comment]
Yes, you are right ;). I came to the same conclusion when thinking about it over night ;).

So maybe working around and using dynamic config parameter hoursOn depending on the prices and battery SoC may take care of what i whish to do :). Maybe i could also use my price monitoring database for this.
